ci/src/mainnet_revisions/mainnet_revisions.py

```diff
@@ -0,0 +1,177 @@
+    )
+
+    if output.returncode != 0:
+        print("Failed to get subnet", subnet_id, "Error:", output.stderr)
+        exit(1)
+
+    response = json.loads(output.stdout)
+    return response["records"][0]["value"]["replica_version_id"]
+
+def get_repo_root() -> str:
+    return subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        text=True,
+        stdout=subprocess.PIPE
+    ).stdout.strip()
+
+def main():
+    """Do the main work."""
+
+    class HelpfulParser(argparse.ArgumentParser):
+        """An argparse parser that prints usage on any error."""
+
+        def error(self, message):
+            sys.stderr.write("error: %s\n" % message)
+            self.print_help()
+            sys.exit(2)
+
+    parser = HelpfulParser()
+
+    parser.add_argument("--verbose", "-v", action="store_true", help="Verbose mode")
+
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    current_nns_version = get_subnet_replica_version(nns_subnet_id)
+    logging.info("Current NNS subnet (%s) revision: %s", nns_subnet_id, current_nns_version)
+    current_app_subnet_version = get_subnet_replica_version(app_subnet_id)
+    logging.info("Current App subnet (%s) revision: %s", app_subnet_id, current_app_subnet_version)
+
+    ic_repo_push_token = os.environ.get("PUSH_TO_IC", "ENV_VAR_NOT_SET")
+    repo = "dfinity/ic"
+
+    repo_root = pathlib.Path(os.environ.get("BUILD_WORKING_DIRECTORY", "NOT_PRESENT"))
+
+    if not repo_root.parent.exists():
+        raise Exception ("Expected dir %s to exist", repo_root.name)
+
+    running_on_ci = os.environ.get("CI") or os.environ.get("GITHUB_ACTION")
+
+    branch = "ic-mainnet-revisions"
+    subprocess.call(["git", "fetch"], cwd=repo_root)
+    if subprocess.call(["git", "checkout", branch], cwd=repo_root) == 0:
+        # The branch already exists, update the existing MR
+        logging.info("Found an already existing target branch")
+    else:
+        subprocess.check_call(["git", "checkout", "-b", branch], cwd=repo_root)
+    subprocess.check_call(["git", "reset", "--hard", "origin/master"], cwd=repo_root)
+
+    update_saved_subnet_version(subnet=nns_subnet_id, version=current_nns_version, repo_root=pathlib.Path(repo_root))
+    update_saved_subnet_version(
+        subnet=app_subnet_id, version=current_app_subnet_version, repo_root=pathlib.Path(repo_root)
+    )
+    git_modified_files = subprocess.check_output(["git", "ls-files", "--modified", "--others"], cwd=repo_root).decode(
+        "utf8"
+    )
+    if SAVED_VERSIONS_PATH in git_modified_files:
+        logging.info("Creating/updating a MR that updates the saved NNS subnet revision")
+        subprocess.check_call(["git", "add", SAVED_VERSIONS_PATH], cwd=repo_root)
+        if running_on_ci:
+            subprocess.check_call(["git", "config", "user.email", "infra@dfinity.org"], cwd=repo_root)
```

Comment:
We use the following email address in [other places](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/dfinity/ic%24+user.email&patternType=keyword&sm=0):
```suggestion
            subprocess.check_call(["git", "config", "user.email", "infra+github-automation@dfinity.org"], cwd=repo_root)
```